src/main/java/com/scaler/lld/design/assignments/singleton/FileBasedConfigurationManagerImpl.java

```diff
@@ -2,49 +2,63 @@
 
 public class FileBasedConfigurationManagerImpl extends FileBasedConfigurationManager {
 
+
+    private static FileBasedConfigurationManagerImpl instance;
+
+    // Private constructor to enforce Singleton pattern
+    private FileBasedConfigurationManagerImpl() {
+    }
     @Override
     public String getConfiguration(String key) {
-        // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'getConfiguration'");
+
+        return properties.getProperty(key);
     }
 
     @Override
     public <T> T getConfiguration(String key, Class<T> type) {
-        // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'getConfiguration'");
+
+        String value = properties.getProperty(key);
+        if (value != null) {
+            return convert(value, type);
+        }
+        return null;
     }
 
     @Override
     public void setConfiguration(String key, String value) {
-        // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'setConfiguration'");
+
+        properties.setProperty(key,value);
     }
 
     @Override
     public <T> void setConfiguration(String key, T value) {
-        // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'setConfiguration'");
+
+        properties.setProperty(key,value.toString());
     }
 
     @Override
     public void removeConfiguration(String key) {
-        // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'removeConfiguration'");
+
+        properties.remove(key);
+
     }
 
     @Override
     public void clear() {
-        // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'clear'");
+
+        properties.clear();
     }
 
     public static FileBasedConfigurationManager getInstance() {
         // TODO Auto-generated method stub
-        return null;
+        if (instance == null) {
+            instance = new FileBasedConfigurationManagerImpl();
+        }
+        return instance;
```

Comment:
This is the correct implementation. However, it does not take into account thread safety. 
Try to implement a thread safe version as well.

Reply:
Done the changes